### PR TITLE
Removing all references to the entrypoint `/all`

### LINKED ISSUE #21: Correct how the all endpoint is specified (or remove it completely from the specs)
What does the /all endpoint presently return?

Without a filter it should return all entries of all entry types, using the jsonapi 'type' member to specify which entry_type each record is.

So, e.g.,
```
   /all?filter=id="_exmpl_42"
```
Looks for any entry in the database with the id _exmpl_42, regardless of which type it is.

### PR DESCRIPTION
**Pull request by giovannipizzi**
Fixes #21

### CODE REVIEW

File: optimade.md [hunk 1]
```diff
@@ -990,16 +961,15 @@ of the provider's other OPTiMaDe databases.
...
 
-## <a name="h.4.6">4.6. Custom Extension Endpoints</a>
+## <a name="h.4.5">4.5. Custom Extension Endpoints</a>
 
 API implementors can provide custom endpoints under the Extensions endpoint.
 They should have the form "&lt;base\_url&gt;/extensions/&lt;custom paths&gt;".
 
 # <a name="h.5">5. API Filtering Format Specification</a>
 
 An OPTIMaDe filter expression is passed in the parameter `filter`
```
Comment (CasperWA) [inline optimade.md]: Randomly spotted that OPTiMaDe is written wrongly here - although it has nothing to do directly with this PR, you should be okay to edit it :)

File: optimade.md [hunk 2]
```diff
@@ -990,16 +961,15 @@ of the provider's other OPTiMaDe databases.
...
-either as an URL query parameter as specified by jsonapi, or as part
-of a POST request as described in [4.3. General entry listing 'All' endpoint](#h.4.3).
+either as an URL query parameter as specified by JSON API.
```
Comment (CasperWA) [inline optimade.md]: Remove "either"

Comment (CasperWA) [reply optimade.md]: And perhaps make JSON API a link to: https://jsonapi.org/format/1.0/#fetching-filtering

Comment (giovannipizzi) [reply optimade.md]: Done, can be re-reviewed now

**Pull request merged**